js_code/offerList.js

```diff
@@ -0,0 +1,516 @@
+            hashtags : ['PC', 'CS', 'MacBook Pro', 'DELL', 's1mple'],
+            discount : '16',
+            validUntil : new Date('2022-01-21T20:12:32'),
+            rating : 3,
+            reviews : ['Xbox is better!'],
+        },
+        {
+            id: '19',
+            description : 'Компьютеры со скидкой 70%',
+            createdAt : new Date('2021-06-26T21:12:11'),
+            link : 'https://PC.pl',
+            vendor : 'Computers - best',
+            photoLink : 'https://lh3.googleusercontent.com/proxy/n4y3C7gnmGV4SOVmkE2DoLGua_DNkt9Nyrnth8-gXnbO_xFewvtRi3ztskcGA4G1mwE45gvXpq_1rWX1VYLXJbh9jJGUc3x0E6exe2T9F1LLN5ZgL4d8KZy4SoaGo9IlVQGFY6mDTznWSptH9XZIf-lK',
+            hashtags : ['PC', 'CS', 's1mple', 'MacBook Pro', 'DELL'],
+            discount : '70',
+            validUntil : new Date('2022-01-21T20:12:32'),
+            rating : 5,
+            reviews : ['Xbox is better!'],
+        },
+        {
+            id: '20',
+            description : 'Компьютеры со скидкой 88%',
+            createdAt : new Date('2021-02-11T20:14:32'),
+            link : 'https://PC.pl',
+            vendor : 'Computers - best',
+            photoLink : 'https://lh3.googleusercontent.com/proxy/n4y3C7gnmGV4SOVmkE2DoLGua_DNkt9Nyrnth8-gXnbO_xFewvtRi3ztskcGA4G1mwE45gvXpq_1rWX1VYLXJbh9jJGUc3x0E6exe2T9F1LLN5ZgL4d8KZy4SoaGo9IlVQGFY6mDTznWSptH9XZIf-lK',
+            hashtags : ['PC', 'CS', 's1mple', 'MacBook Pro', 'DELL'] ,
+            discount : '88',
+            validUntil : new Date('2022-01-18T22:19:42'),
+            rating : 1,
+            reviews : ['Xbox is better!'] ,
+        }
+    ]
+
+  function getAds(skip = 0, top = 10, filterConfig){
+    if (typeof skip !== 'number' || typeof top !== 'number') {
+        console.log('Error with inputting types!');
+        return;
+    }
+
+    let returningAdsList = adList;
+
+    if (filterConfig){
+        for (let param in filterConfig){
+            if (param === 'hashtags'){
+                for (let i = 0; i < filterConfig.hashtags.length; i++){
+                    returningAdsList = returningAdsList.filter(item => item.hashtags.includes(filterConfig.hashtags[i]));
+                }
+            }
+            else if (param === 'vendor'){
+                returningAdsList = returningAdsList.filter(item => item.vendor === filterConfig.vendor);
+            }
+            else if (param === 'dateFrom'){
+                returningAdsList = returningAdsList.filter(item => item.createdAt >= filterConfig.dateFrom);
+            }
+            else if (param === 'dateTo'){
+                returningAdsList = returningAdsList.filter(item => item.createdAt < filterConfig.dateTo);
+            }
+        }
+    }
+
+    returningAdsList.sort(comparator);
+    return returningAdsList.slice(skip, skip + top);
```

Comment:
Looks completely the same as other students' method (even string you log to console), so I'll decrease score for each of you who used this code.